src/main/java/me/zeroeightsix/kami/util/astaralg/Astarpathfinder.java

```diff
@@ -0,0 +1,650 @@
+
+        //for testing!!! can be removed after
+        MessageSendHelper.sendChatMessage("Checking if blockstuck:" + (condition ? (TextFormatting.RED + " Stuck") : (TextFormatting.GREEN + " Only Damage")));
+        if (condition) {
+            blockstuck = true;
+            mc.timer.tickLength = 50.0f;
+            mc.player.motionZ = 0;
+            mc.player.motionX = 0;
+            mc.player.motionY = -0.4;
+            mc.player.capabilities.isFlying = false;
+            mc.player.capabilities.setFlySpeed(0.05f);
+        }
+    }
+
+    public static void update(PlayerTravelEvent event) {
+        currentEvent = event;
+        try {
+            switch (state) {
+                case START_FLYING:
+                    startFlying();
+                    break;
+                case PATHFIND:
+                    pathfind();
+                    break;
+                case GOTO_TRACK:
+                    gotoTrack();
+                    break;
+                case LAND_SAVE:
+                    landSave();
+                    break;
+            }
+        } catch (Exception e) {
+            MessageSendHelper.sendChatMessage("Error occurred send logs in discord ");
+            //i mean try catch barely uses performance
+            e.printStackTrace();
+        }
+
+    }
+
+
+    private static void landSave() {
+        if (mc.player.onGround || landed) {
+            landed = true;
+            mc.timer.tickLength = 50.0f;
+            if (Math.abs(getPlayerBlockpos().getX() - stuckPos.getX()) < 3 && Math.abs(getPlayerBlockpos().getZ() - stuckPos.getZ()) < 3) {
+                BaritoneAPI.getProvider().getPrimaryBaritone().getPathingBehavior().cancelEverything();
+                buildstate = TrackState.GOTO_BUILD;
+                state = State.GOTO_TRACK;
+                currentDirection = direction;
+                landed = false;
+                blockstuck = false;
+
+            } else {
+                if (BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().getGoal() == null) {
+                    BaritoneAPI.getProvider().getPrimaryBaritone().getCustomGoalProcess().setGoalAndPath(new GoalXZ(stuckPos.getX(), stuckPos.getZ()));
+                }
+            }
+        } else {
+            if (!BlockUtils.checkForLiquid()) {
+                // if blockstuck let vanilla mechanics run (i.e dont cancel the event) seems to work fine
+                if (!blockstuck)
+                    elytraFlight.landing(currentEvent);
+            } else {
+                currentDirection = direction + 4 % 8;
+                pathfind();
+            }
+        }
+//        the plan was to do something like this:
+//        switch (landstate) {
+//            case FLY_TO_OBSTACLE:
+//                //TODO: Fly to the obstacle
+//                break;
+//            case LAND:
+//                //TODO:  find save landing spot and fly there
+//                break;
+//        }
```

Comment:
is this not happening or postponed?

Reply:
postponed i'd say. maby ill do something completely different idk yet. I can remove it + the landstate enum if you want for now